### ISSUE TITLE
Use multi-release jar

### ISSUE THREAD

**Issue by Thihup** [tginsberg/gatherers4j#25]
I'd like to know if this project will use multi-release jar to have the support from Java 22 to Java 24 (as it is not possible to use classes compiled with --enable-preview from another Java version)?

**Comment by tginsberg** [comment]
I hadn't considered it, is there a reason why you'd need support for 22 or 23 once 24 is out? They'll all have EOL'd, and 25 is LTS.

My rough plan had been to stick with the current six-month release until we hit an LTS release and then stay there for a while. So, keep on 22 September 2024 when 23 comes out. Stay on 23 until March 2024 when 24 comes out. And then move to 25 in September 2024 and stay there until there is a compelling version to upgrade. I don't really want to be in the business of providing builds for JDK versions that have EOL'd if I can avoid it. Besides, it looks like [Stream Gatherers will finalize in JDK 24](https://openjdk.org/jeps/485) (which will EOL when 25 comes out). In terms of the `--enable-preview` flag, that will go away once 24 comes out (see the link!).

**Comment by tginsberg** [comment]
I'll close this, but feel free to comment if I'm not addressing your point directly.

Here's my plan for this library, as it relates to JDK versions:

TL;DR: Move the minimum Java version up until we hit an LTS version (25) and stay there until there is a compelling reason to move from it (Java 29 comes out, Gatherers API has material changes, etc), at which point I'll evaluate multi-release jars to ease transitions.

| Date  | Latest Java Version | Minimum Version for Gatherers4J | Notes |
| ------------- | ------------- |--------------------------------|--|
| Today  | 22  | 22 | Gatherers API in Preview |
| September 2024 | 23  | 23 | Gatherers API in Preview |
| March 2025  | 24  | 24 |  Gatherers API is Public |
| September 2025 | 25  | 25 | Java 25 is LTS |
| March 2026  | 26  | 25  | Stay on LTS |
